samples/back_stack/src/main/java/com/badoo/ribs/samples/back_stack/parent/ParentPresenter.kt

```diff
@@ -0,0 +1,88 @@
+package com.badoo.ribs.samples.back_stack.parent
+
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.android.subscribe
+import com.badoo.ribs.core.plugin.ViewAware
+import com.badoo.ribs.routing.source.backstack.BackStack
+import com.badoo.ribs.routing.source.backstack.operation.newRoot
+import com.badoo.ribs.routing.source.backstack.operation.push
+import com.badoo.ribs.routing.source.backstack.operation.pushOverlay
+import com.badoo.ribs.routing.source.backstack.operation.replace
+import com.badoo.ribs.routing.source.backstack.operation.singleTop
+import com.badoo.ribs.samples.back_stack.parent.routing.ParentRouter
+
+typealias ViewEventContent = ParentView.Event.RoutingContent.Content
+typealias ViewEventOverlay = ParentView.Event.RoutingOverlay.Overlay
+
+typealias RouterConfigurationContentA = ParentRouter.Configuration.Content.A
+typealias RouterConfigurationContentB = ParentRouter.Configuration.Content.B
+typealias RouterConfigurationContentC = ParentRouter.Configuration.Content.C
+typealias RouterConfigurationContentD = ParentRouter.Configuration.Content.D
+typealias RouterConfigurationOverlayE = ParentRouter.Configuration.Overlay.E
+typealias RouterConfigurationOverlayF = ParentRouter.Configuration.Overlay.F
+
+interface ParentPresenter {
+
+    fun onEventClicked(event: ParentView.Event)
+
+}
+
+internal class ParentPresenterImpl(
+    private val backStack: BackStack<ParentRouter.Configuration>
+) : ViewAware<ParentView>,
+    ParentPresenter {
+
+    private var view: ParentView? = null
+
+    override fun onViewCreated(view: ParentView, viewLifecycle: Lifecycle) {
+        super.onViewCreated(view, viewLifecycle)
+        viewLifecycle.subscribe(
+            onCreate = {
+                this.view = view
+                this.view?.setBackStack(getBackStackString())
+            },
+            onDestroy = { this.view = null }
+        )
+
+        backStack.observe { routingHistory ->
+            if (routingHistory is BackStack.State) this.view?.setBackStack(getBackStackString())
+        }
+    }
+
+    override fun onEventClicked(event: ParentView.Event) {
+        when (event) {
+            is ParentView.Event.RoutingContent.Pop -> backStack.popBackStack()
```

Comment:
Suggestion:
Import ParentView.Event
Rename RoutingContent -> Content, RoutingOverlay -> Overlay

Reply:
We already have enums in that class with that same name, what do you think about RoutingContent -> ContentAction, RoutingOverlay -> OverlayAction?

Reply:
I suggest moving out and unifying the inner enums that refer to letters, into a single enum containing both.